### ISSUE TITLE
txReceipt design changes

### ISSUE THREAD

**Issue by alextsg** [MetaMask/Design#46]
Each transaction will now have transaction receipt data upon confirmation which will give us some additional information for each transaction. For example, 1) in terms of gas we're currently only showing the gas price and the gas limit in the breakdown, but we'll now know how much gas was actually used for the transaction, and 2) we show a green "Confirmed" label for transactions that were included in a block, but we'll now know if a transaction has failed on-chain as well.

I'm thinking that some design changes might be needed to present this new information to users. Looking at the transaction breakdown:
![image](https://user-images.githubusercontent.com/8051479/46697550-baf0ac00-cbc9-11e8-87d4-afab9d943bfd.png)
Where should the actual gas used be included? Maybe as a row underneath the "Gas Price"? Also, the "Total" value is calculated by multiplying the gas price by the gas limit, when it should actually be a product of the gas price and the gas used.

For the transaction status labels, a green "Confirmed" currently implies a successful transaction, when what it actually means is that it was included in a block and it's possible that the transaction failed on-chain. We can introduce 2 new states that represent "Confirmed and succeeded on-chain" and "Confirmed and failed on-chain". Should we use a new term instead of "Confirmed"? And maybe use a different color instead of green? (I guess we could even use the "Pending" orange since the transaction is still in flux) It might be a good idea to have description text for each status label that we can show as a tooltip when hovering over the status.

In summary, what this ticket is requesting for is 1) where should we show the gas used, and 2) a clear categorization of how we want to represent the different statuses of a transaction, ie. state name, color, and description.

@frankiebee Am I missing anything here, and is anything inaccurate?

**Comment by cjeria** [comment]
It's fine to show more data, but i don't think we need to be so specific as to show whether the transaction successfully failed on-chain or before that point. Most average users only care whether the transaction was received by the contract or recipient. Perhaps it's valuable data for dissecting issues with transactions, however I'm confident that an average non-technical user really understand or cares to understand these lower level concepts (ie, most users don't understand the difference between a seedphrase and private keys). That said, keeping it generalized (pass/fail) regardless of whether it failed before it made into a block or it failed on-chain is enough to keep users informed at a high-level that the transaction simply failed.  Also, wouldn't this lower level data of a transaction be available in etherscan? 

Regarding the actual gas used, that's the data the **should** be presented to the user, not the stale data (estimate gas data?). Those should be overwritten with the actual values.

@bdresser @danfinlay your thoughts?

**Comment by alextsg** [comment]
The main reason we'd want to show those states would be to avoid confusion like this: https://github.com/MetaMask/metamask-extension/issues/5117
We show "Confirmed" because the transaction has made it into a block, and from the UI the user thinks that the transaction has gone through. What's really happened is that the transaction failed on-chain, no transfer has taken place, and the user and recipient's token balances are unchanged.

**Comment by bdresser** [comment]
Agree with @cjeria that we should stick to `Confirmed` / `Failed` on the home screen, and I think on-chain failures should show up as `Failed` as well. Anecdotal evidence says that's what users expect when something fails on-chain. 

We can use the Activity Log to explain what has actually happened (that's what it's for!) So an on-chain failure could show up as

On home screen, `Failed`
In tx detail view:
- tx created with a value of .5 ETH
- tx submitted on 10/09...
- tx confirmed on 10/09...
- tx failed with error `Error Encounter during Contract Execution [out of gas]`

Regarding gas: I think keeping the gas options the user submitted with is valuable -- what if they liked the confirmation time and want to repeat the same settings? 

I would suggest just adding another row `Gas Used` and then updating the total with the more accurate info once we have it.

**Comment by alextsg** [comment]
Sorry yeah I think I just misunderstood @cjeria 's point. Yeah that makes sense, to use our existing labels but applying "Failed" to on-chain failures as well.

**Comment by bdresser** [comment]
@alextsg i'm seeing part of these changes in prod -- had a tx fail on chain and it was marked "Failed" in my tx history, which is great.

But the Activity Log didn't update or have any additional info -- bug, or not implemented yet? want me to make an issue or you got this handled?

<img width="2245" alt="screen shot 2018-10-22 at 12 06 39 pm" src="https://user-images.githubusercontent.com/1016190/47313190-25a2de00-d5f3-11e8-9482-80878671f87d.png">

**Comment by alextsg** [comment]
@bdresser Yeah I just need to implement that - the data for the on-chain failure is coming from a difference source from the activity log so I just need to consolidate the information. Will have that done shortly so you don't need to create a new issue. Thanks.

**Comment by alextsg** [comment]
@bdresser PR for the above: https://github.com/MetaMask/metamask-extension/pull/5581
